services/shhext/chat/sharedsecret/persistence.go

```diff
@@ -0,0 +1,125 @@
+package sharedsecret
+
+import (
+	"database/sql"
+	"strings"
+)
+
+type PersistenceService interface {
+	Add(identity []byte, secret []byte, installationID string) error
+	Get(identity []byte, installationIDs []string) (*Response, error)
+	All() ([][][]byte, error)
```

Comment:
If that's ok, we should put a comment explaining what is in each level of this multidimensional slice.